### PR TITLE
Support osx-arm64

### PR DESCRIPTION
**Pull request by longfin**
This PR adds builds for Apple Silicon (a.k.a. M1/M2) to GitHub Releases and NPM. 

I'm not sure this is all that needs to be changed due to a lack of context. so please let me know if there's something else that should be considered.

### CODE REVIEW

**Comment by dahlia** [comment]
The [built binary][1] actually does not work on Apple silicon Macs:

```console
$ 7z x dist-bin.zip
$ tar xvfJ planet-0.43.0-dev.20221001102841-osx-arm64.tar.xz 
./Libplanet.Extensions.Cocona.pdb
./Libplanet.Net.pdb
./Libplanet.Net.xml
./Libplanet.RocksDBStore.pdb
./Libplanet.Stun.pdb
./Libplanet.pdb
./Libplanet.xml
./libgflags.2.2.dylib
./liblz4.1.dylib
./librocksdb.dylib
./libsnappy.1.dylib
./libzstd.1.dylib
./planet
./planet.pdb
$ file *.dylib planet
libgflags.2.2.dylib: Mach-O 64-bit dynamically linked shared library arm64
liblz4.1.dylib:      Mach-O 64-bit dynamically linked shared library arm64
librocksdb.dylib:    Mach-O 64-bit dynamically linked shared library arm64
libsnappy.1.dylib:   Mach-O 64-bit dynamically linked shared library arm64
libzstd.1.dylib:     Mach-O 64-bit dynamically linked shared library arm64
planet:              Mach-O 64-bit executable arm64
$ ./planet 
Killed: 9
```

It's the exactly same issue that made me to give up the [previous attempt][2]…

See also: <https://github.com/planetarium/libplanet/issues/2280#issuecomment-1238870064>

[1]: https://github.com/planetarium/libplanet/suites/8565116426/artifacts/382975775
[2]: https://github.com/planetarium/libplanet/pull/1650

**Comment by longfin** [comment]
I succeed to run an executable built from CI with an adhoc code signing.

```bash
bash-3.2$ ./planet
Killed: 9

longfin@longfinui-MacBookAir osx-arm64 % codesign -s - -f planet
longfin@longfinui-MacBookAir osx-arm64 % ./planet
objc[41846]: Class PlaceholderObject is implemented in both /Users/longfin/Downloads/osx-arm64/planet (0x1017393f0) and /Users/longfin/Downloads/osx-arm64/libSystem.Native.dylib (0x122140600). One of the two will be used. Which one is undefined.
Usage: planet [command]

planet

Commands:
  apv
  key
  mpt
  store
  stats

Options:
  -h, --help    Show help message
  --version     Show version
```

I think that [Gatekeeper](https://support.apple.com/en-US/guide/security/sec5599b66df/web) may disturb running... (idk why it's okay on x86-64 binary, but I guess it may be related to Rosseta2 🤔 )

Anyway, we need an integration for codesigning. I'll work on that when available.

**Comment by longfin** [comment]
- After macOS 11 on Apple Silicon(arm64), Apple seems to require a codesign more strictly.
- Since [Adhoc signing](https://github.com/planetarium/libplanet/pull/2365/#issuecomment-1266343562) is only valid in the environment of the user who signed it, it cannot be used for CI.
- Some package managers (e.g., Homebrew, `dotnet tools`) seemed to be prepared for that requirement through Adhoc code signing. ([1], [2])
  - so If `planet` is to be installed only through such channels, we can omit code signing.
  - In that case, our downloader(i.e., `download.mjs`) also should be adjusted.
- On the other hand, in order to fundamentally eliminate concerns, it is also a possible option for someone in the project to represent and register for the Apple Developer Program and feed to CI.
  - However, who is appropriate to represent this needs to be discussed and may take time.

Unless otherwise noted, it seems like a viable option right away, so I'm working towards adding ad hoc signing.

@planetarium/libplanet Any ideas? 

[1]: https://github.com/Homebrew/brew/issues/9082
[2]: https://github.com/dotnet/runtime/issues/52335

**Comment by dahlia** [comment]
@longfin IMHO, it would be the quickest and the most approachable at present to make *download.mjs* to ad-hoc sign it immediately after downloading the executable on macOS.

**Comment by longfin** [comment]
Added adhoc signing on https://github.com/planetarium/libplanet/pull/2365/commits/670f8f30b704361cf145eede41dcb8d786d5fbf1.

PTAL 🙏  @dahlia (and other Apple Silicon users)

File: Libplanet.Tools/download.mjs [hunk 1]
```diff
@@ -119,7 +143,26 @@ Report this issue from https://github.com/planetarium/libplanet/issues
 ${err}
 -------------------------------------------------------------------------------
     `);
-    throw e;
+    throw err;
+  }
+  if (platform === "darwin" && arch === "arm64")
```
Comment (dahlia) [inline Libplanet.Tools/download.mjs:148]: Wouldn't it better to ad-hoc sign the executable on Intel Macs too?

**Review by dahlia — APPROVED** [review]

File: Libplanet.Tools/download.mjs (hunk 1 above)
Comment (longfin) [reply Libplanet.Tools/download.mjs:148]: makes sense, but I didn't have Intel Mac to test😅 .

Comment (dahlia) [reply Libplanet.Tools/download.mjs:148]: Okay, never mind then.  Anyway, could you resolve the conflict?

**Review by dahlia — APPROVED** [review]

**Review by dahlia — CHANGES_REQUESTED** [review]
Oh, we almost forgot about docs.  Could you adjust the following part from *Libplanet.Tools/README.md*?  It would be shown on [docs.libplanet.io](https://docs.libplanet.io/) as well.

https://github.com/planetarium/libplanet/blob/dca2386f7ef33807491a2650dbb3e02dede576be/Libplanet.Tools/README.md#L45-L48

**Comment by longfin** [comment]
> Oh, we almost forgot about docs. Could you adjust the following part from Libplanet.Tools/README.md? It would be shown on [docs.libplanet.io](https://docs.libplanet.io/) as well.

Added https://github.com/planetarium/libplanet/pull/2365/commits/c840c986168bdda2c4d9b140c8514b253bae752e, PTAL @dahlia 🙏

**Review by dahlia — APPROVED** [review]

**Review by dahlia — APPROVED** [review]

**Comment by greymistcube** [comment]
@dahlia @longfin 
Eh... this breaks linux-mono-build (not tests). Can you take another look?

**Comment by longfin** [comment]
> @dahlia @longfin Eh... this breaks linux-mono-build (not tests). Can you take another look?

Sure, I'll check again now.

**Comment by longfin** [comment]
> > @dahlia @longfin Eh... this breaks linux-mono-build (not tests). Can you take another look?
> 
> Sure, I'll check again now.

Opened the new PR(#2462) about that. cc @greymistcube @dahlia

**Pull request merged**